Flaky test due to negative ENR sequence number

## What was wrong?

```
make_request = <function _make_request.<locals>.make_request at 0x7f70e8769b80>
w3 = <web3.main.Web3 object at 0x7f70e940f370>
new_enr = <[SerializationError('Cannot serialize negative integers') raised in repr()] ENR object at 0x7f70e8b15820>

    @pytest.mark.trio
    async def test_v51_rpc_set_enr_web3_duplicate(make_request, w3, new_enr):
        response = await trio.to_thread.run_sync(w3.discv5.set_enr, repr(new_enr))
        assert response is None
    
        new_enr._sequence_number = new_enr._sequence_number - 1
        with pytest.raises(Exception, match="Invalid ENR"):
>           await trio.to_thread.run_sync(w3.discv5.set_enr, repr(new_enr))
E           AssertionError: Regex pattern 'Invalid ENR' does not match 'Cannot serialize negative integers'.

make_request = <function _make_request.<locals>.make_request at 0x7f70e8769b80>
new_enr    = <[SerializationError('Cannot serialize negative integers') raised in repr()] ENR object at 0x7f70e8b15820>
response   = None
w3         = <web3.main.Web3 object at 0x7f70e940f370>
```

## How can it be fixed?

Need to ensure that `new_enr` has a sequence number of `>=1`

--- a/tests/core/v5_1/test_v51_rpc_handlers.py
+++ b/tests/core/v5_1/test_v51_rpc_handlers.py
@@ -114,7 +114,7 @@ def bob_node_id_param_w3(request, alice, bob, bob_network):
 
 @pytest.fixture
 def new_enr():
-    return ENRFactory()
+    return ENRFactory(sequence_number=secrets.randbelow(100) + 1)
 
 
 @pytest.mark.trio
@@ -217,7 +217,9 @@ async def test_v51_rpc_set_enr_web3(make_request, w3, new_enr):
 
 
 @pytest.mark.trio
-async def test_v51_rpc_set_enr_web3_duplicate(make_request, w3, new_enr):
+async def test_v51_rpc_set_enr_web3_with_invalid_sequence_number(
+    make_request, w3, new_enr
+):
     response = await trio.to_thread.run_sync(w3.discv5.set_enr, repr(new_enr))
     assert response is None
 

